linker/shared/src/test/scala/org/scalajs/linker/testutils/TestIRBuilder.scala

```diff
@@ -71,7 +71,7 @@ object TestIRBuilder {
   }
 
   def mainMethodDef(body: Tree): MethodDef = {
-    val stringArrayType = ArrayType(ArrayTypeRef("T", 1))
+    val stringArrayType = ArrayType(ArrayTypeRef(ClassRef(BoxedStringClass), 1))
```

Comment:
I also do not understand this :)

Reply:
Hum?

1. `"T"` -> `ClassRef("T")` to adapt to the new arguments of `ArrayTypeRef`
2. Replace the literal `"T"` by the named constant `BoxedStringClass` which is equal to `"T"`

Perhaps you thought `"T"` was for *primitive* strings rather than "boxed" strings? It's a historical anomaly that `"T"` stands for `java.lang.String` rather than the primitive `StringType`. It was like that for shortening encoded names as much as possible. This thing will disappear once I get to delaying mangling to the Emitter, which should hopefully come after this PR (this PR came to fix difficulties I encountered when trying to implement the delayed mangling, actually).